javers-core/src/main/java/org/javers/core/metamodel/type/ManagedClassFactory.java

```diff
@@ -41,19 +38,19 @@
     }
 
     public ClientsDomainClass inferFromAnnotations(Class javaClass) {
-        List<Property> properties = propertyScanner.scan(javaClass);
-
-        Property foundIdProperty = findIdProperty(properties);
-
-        if (foundIdProperty != null) {
-            return new Entity(javaClass, properties, foundIdProperty);
+        ClientsClassDefinition clientsClassDefinition = classAnnotationsScanner.scanAndInfer(javaClass);
```

Comment:
Nie. Nie wiedziałem dlaczego jest tworzona encja w tej linii. Bo w tej samej klasie w funkcji jest tworzona Encja w funkcji create(ClientsClassDefinition clientsClassDefinition) w trochę inny sposób:
if (clientsClassDefinition instanceof EntityDefinition) {
              return create((EntityDefinition) clientsClassDefinition);
}

tylko nie wiem dlaczego tak to jest zrobione. Gdy bym nie omijał tego kodu do tworzenia ShallowReference nie doszło by.